### PR TITLE
Support OAuth Protected Engine APIs

### PR DESCRIPTION
**Pull request by lukasniemeier-zalando**
With this change OAuth protected APIs of the engine are supported (when deployed as a standalone web application next to the cockpit).

### CODE REVIEW

File: cockpit-client.py [hunk 1]
```diff
@@ -91,7 +129,7 @@ def get_failed_jobs(self, message_filter=None, from_timestamp=None, to_timestamp
 
     def show_failed_jobs(self, message_filter=None, from_timestamp=None, to_timestamp=None, process_instance_id=None):
         row_format = \
-                '{processEngine:8}{incidentTimestamp:24}{processDefinitionKey:24}{activityId:40}{processInstanceId:64}{exceptionMessage}'
+                '{processEngine:8}{incidentTimestamp:24}{processDefinitionKey!s:24}{activityId:40}{processInstanceId:64}{exceptionMessage}'
```
Comment (lukasniemeier-zalando) [inline cockpit-client.py]: I've encountered missing `processDefinitionKey`s, I haven't look into it that much.

Comment (jhorstmann) [inline cockpit-client.py]: Sounds strange and should not happen. What does `!s` do then, print `None`?

**Comment by jhorstmann** [comment]
:+1:

File: cockpit-client.py (hunk 1 above)
Comment (lukasniemeier-zalando) [inline cockpit-client.py]: Yes, forces a conversion to `String`

**Pull request merged**